tests/mir-opt/matches_reduce_branches.rs

```diff
@@ -66,44 +70,172 @@ fn match_nested_if() -> bool {
     val
 }
 
+// # Fold switchInt into IntToInt.
+// To simplify writing and checking these test cases, I use the first character of
+// each case to distinguish the sign of the number:
+// 'u' for unsigned, '_' for negative, and 'o' for non-negative.
+// Followed by a decimal number, and add the corresponding radix representation.
+// For example, o127_0x7f represents 127i8, and _1_0xff represents -1i8.
+
+// ## Cast but without numeric conversion.
+
 #[repr(u8)]
 enum EnumAu8 {
-    A = 1,
-    B = 2,
+    u0_0x00 = 0,
+    u127_0x7f = 127,
+    u128_0x80 = 128,
+    u255_0xff = 255,
+}
+const _: () = assert!(127u8 == 0x7fu8 && 128u8 == 0x80u8 && 255u8 == 0xffu8);
```

Comment:
Is this necessary?

Reply:
I just want to avoid some careless mistakes. :p
I think this will help with the review as well.